### PR TITLE
RDM-5499 - handling duplicate / existing user scenario.

### PR DESCRIPTION
**Pull request by smathangi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-5499
### Change description ###

handling duplicate / existing user scenario.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java [hunk 1]
```diff
@@ -48,7 +52,8 @@ public UserProfileRepository(JurisdictionRepository jurisdictionRepository,
      */
     public UserProfile createUserProfile(UserProfile userProfile, final String actionedBy) {
         if (null != findEntityById(userProfile.getId(), actionedBy, false)) {
-            throw new BadRequestException("User already exists with Id " + userProfile.getId());
+            LOG.warn("User already exists with id:{} so not creating again", userProfile.getId());
+            return userProfile;
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java]: Sateesh, maybe I misunderstood, but I thought we would overwrite in case the user already exists, rather than ignoring the new request.

File: src/test/java/uk/gov/hmcts/ccd/endpoint/userprofile/UserProfileEndpointIT.java [hunk 2]
```diff
@@ -210,10 +210,9 @@ public void createExistingUserProfile() throws Exception {
                         .content(mapper.writeValueAsBytes(userProfile)))
                 .andReturn();
 
-        assertEquals("Unexpected response status", 400, mvcResult.getResponse().getStatus());
-        assertEquals("Unexpected response message",
-                     "User already exists with Id " + USER_ID_1,
-                     mvcResult.getResponse().getContentAsString());
+        assertEquals(201, mvcResult.getResponse().getStatus());
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/userprofile/UserProfileEndpointIT.java]: if we want to go with the ignore solution, I think we should not return 201 in this case. Can we return a more suitable response code and message to indicate that the request was ignored?

File: src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java [hunk 3]
```diff
@@ -64,10 +69,14 @@ public UserProfile createUserProfile(UserProfile userProfile, final String actio
         userProfileAuditEntityRepository.createUserProfileAuditEntity(userProfile,
                                                                       CREATE,
                                                                       actionedBy,
-                                                                      userProfile.getWorkBasketDefaultJurisdiction());
+                                                                      getFirstJurisdiction(userProfile));
         return UserProfileMapper.entityToModel(userProfileEntity);
     }
 
+    private String getFirstJurisdiction(UserProfile userProfile) {
+        return userProfile.getJurisdictions().stream().findFirst().map(Jurisdiction::getId).orElse(null);
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java]: for my knowledge, what's the rationale of this change?

Comment (smathangi) [reply src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java]: there is no WorkBasketDefaultJurisdiction so.. audit with first Jurisdiction.

File: src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java (hunk 1 above)
Comment (smathangi) [reply src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java]: Yeah... that's what PUT is doing atm but it is tightly coupled with workbasket things  :(

File: src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java [hunk 4]
```diff
@@ -39,35 +41,47 @@ public UserProfileRepository(JurisdictionRepository jurisdictionRepository,
...
-            JurisdictionEntity jurisdictionEntity = jurisdictionRepository.findEntityById(jurisdiction.getId());
-            if (null != jurisdictionEntity) {
-                existingJurisdictions.put(jurisdiction.getId(), jurisdictionEntity);
-            }
+        UserProfileEntity userProfileEntity = findEntityById(userProfile.getId(), actionedBy, false);
+        if (userProfileEntity == null) {
+            auditAction = CREATE;
+            userProfileEntity = UserProfileMapper.modelToEntity(userProfile, existingJurisdictions);
+        } else {
+            auditAction = UPDATE;
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java]: in case of update, can't we reuse the updateUserProfile method?

Comment (smathangi) [reply src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java]: existing updateUserProfile(..) method has workbasket checks inside. If i refactor that too then scope of testing will increase..
